inference-engine/src/multi_device/multi_device_exec_network.cpp

```diff
@@ -60,71 +90,243 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const DeviceMap<Infer
+    _devicePriorities = metaDevices;
+
+    std::vector<DeviceInformation> needLoadDevices;
+
+    // check if have cpu device
+    const auto CPUIter = std::find_if(metaDevices.begin(), metaDevices.end(),
+                                      [=](const DeviceInformation& d)->bool{return d.deviceName.find("CPU") != std::string::npos;});
+    if (CPUIter != metaDevices.end()) {
+        _cpuDevice = *CPUIter;
+        _config.insert(_cpuDevice.config.begin(), _cpuDevice.config.end());
+        needLoadDevices.push_back(_cpuDevice);
+        _cpuFuture = _cpuPromise.get_future();
+    }
+
+    // get accelerator device, like GPU
+    auto networkPrecision = GetNetworkPrecision(network);
+    _acceleratorDevice = _multiPlugin->SelectDevice(metaDevices, networkPrecision);
+    bool isAccelerator =
+        _acceleratorDevice.deviceName.find("CPU") == std::string::npos;
+    if (isAccelerator) {
+        _config.insert(_acceleratorDevice.config.begin(), _acceleratorDevice.config.end());
+        needLoadDevices.push_back(_acceleratorDevice);
+        _acceleratorFuture = _acceleratorPromise.get_future();
+    }
+
+    if (needLoadDevices.size() == 0) {
+        IE_THROW() << "No device set";
+    }
+
+    std::vector<Task> loads;
+    for (auto& p : needLoadDevices) {
+        // initialize these containers firstly to aovid insert operation in threads
+        _idleWorkerRequests[p.deviceName];
+        _workerRequests[p.deviceName];
+        _inferPipelineTasksDeviceSpecific[p.deviceName] = NULL;
+        const auto device = p.deviceName;
+        // will not wait for accelerator network load task,
+        // so some parameters need to be transferred by value
+        loads.push_back([&, modelPath, network, device]() {
+            SoExecutableNetworkInternal executableNetwork;
+            if (!modelPath.empty()) {
+                executableNetwork = core->LoadNetwork(modelPath, device, {});
+            } else {
+                executableNetwork = core->LoadNetwork(network, device, {});
+            }
+
+            GenerateWorkers(device, executableNetwork);
+
+            if (device.find("CPU") == std::string::npos) {
+                _alreadyActualNetwork = true;
+                _acceleratorPromise.set_value(executableNetwork);
+            } else {
+                _cpuPromise.set_value(executableNetwork);
+            }
+        });
+    }
+
+    // will not wait for accelerator network load task
+    // so the exector can't be destoried before the task finished
+    // so use executor as a member of MultiDeviceExecutableNetwork
+    _executor = InferenceEngine::ExecutorManager::getInstance()->getIdleCPUStreamsExecutor(
+            IStreamsExecutor::Config{"AutoDeviceAsyncLoad",
+            static_cast<int>(std::thread::hardware_concurrency()) /* max possible #streams*/,
+            1 /*single thread per stream*/,
+            IStreamsExecutor::ThreadBindingType::NONE});
+
+    for (auto& task : loads) {
```

Comment:
why do we put the tasks to the vector and then iterate over the vector here?
maybe start the task right in the previous loop?